usr/src/lib/libzfs_core/common/libzfs_core.c

```diff
@@ -171,27 +171,32 @@ lzc_ioctl(zfs_ioc_t ioc, const char *name,
 }
 
 int
-lzc_create(const char *fsname, enum lzc_dataset_type type, nvlist_t *props)
+lzc_create(const char *fsname, enum lzc_dataset_type type, nvlist_t *props,
+    nvlist_t *hidden_args)
```

Comment:
While I'm OK with changing the function signature, and we have done it before, I have also seen that it can become a big pain for consumers.  I will be OK either way, but I'd encourage you to seriously consider adding another func that takes the additional argument.  You could call it lzc_create2() -- icky but sustainable.